UPGRADING.md

```diff
@@ -1,6 +1,21 @@
 Upgrading Hashie
 ================
 
+### Upgrading to HEAD
+
+One of the features of Hashie::Mash was broken in MRI Ruby versions 2.2.0, 2.2.1, and 2.2.2.  @michaelherold filed the [bug with ruby-core](https://bugs.ruby-lang.org/issues/10985) and it was fixed in the Ruby 2.2.3 release.  The travis build matrix has been updated to 2.2.3 now that it has been released.
+
+The broken behavior was (in `spec/hashie/mash_spec.rb`):
+
+```ruby
+it 'is able to access an unknown suffixed key as a method' do
+  # See https://github.com/intridea/hashie/pull/285 for more information
+  %w(= ? ! _).each do |suffix|
+    expect(subject.method(:"xyz#{suffix}")).to_not be_nil
+  end
+end
+```
```

Comment:
As for the upgrading text... I am having a terrible time understanding the issue.  I am unable to reproduce the bug on Ruby 2.0.0 or 2.1.7 outside of the spec file, and the `spec/hashie/mash_spec.rb` run differently when running the whole suite (where it passes with on pending spec), versus running just the single file (where I get another failure in addition to the pending spec).

Ruby 2.0.0
```
ruby-2.0.0@foss
∴ irb
>> require "bundler/setup"
=> true
>> require "hashie"
=> true
>> a = Hashie::Mash.new
=> #<Hashie::Mash>
>> a.method(:"xyz=")
=> #<Method: Hashie::Mash#xyz=>
```
Ruby 2.1.7
```
ruby-2.1.7@foss
∴ irb
>> require "bundler/setup"
=> true
>> require "hashie"
=> true
>> a = Hashie::Mash.new
=> #<Hashie::Mash>
>> a.method(:"xyz=")
=> #<Method: Hashie::Mash#xyz=>
```
Ruby 2.2.0
```
ruby-2.2.0@foss
∴ irb
>> require "bundler/setup"
=> true
>> require "hashie"
=> true
>> a = Hashie::Mash.new(a: :b)
=> #<Hashie::Mash a=:b>
>> a.method(:"a=")
=> #<Method: Hashie::Mash#a=>
>> a.method(:"a=").nil?
=> false
>> a.method(:"xyz=").nil?
=> false
>> a.method(:"xyz?").nil?
=> false
>> a.method(:"xyz!").nil?
=> false
>> a.method(:"xyz_").nil?
=> false
>> a.method(:"xyz_").class
=> Method
>> a.method(:"xyz_")
=> #<Method: Hashie::Mash#xyz_>
>> RUBY_VERSION
=> "2.2.0"
```

I am going to just remove the upgrading text.  I don't even understand the problem, so I really should be the one documenting it.